### PR TITLE
test: Refactor group user reports tests

### PR DESCRIPTION
**Pull request by lynnagara**
Remove the base test class UserReportEnvironmentTestCase since we only use
this in one place now.
This is a follow up to https://github.com/getsentry/sentry/pull/14293

### CODE REVIEW

File: tests/sentry/api/endpoints/test_group_user_reports.py [hunk 1]
```diff
@@ -1,15 +1,67 @@
...
         )
 
+    def create_environment(self, project, name):
+        env = Environment.objects.create(
+            project_id=project.id,
+            organization_id=project.organization_id,
+            name=name,
+        )
+        env.add_project(project)
+        return env
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_group_user_reports.py]: This looks very similar to the `create_environment()` method that sentry/testutils/factories exposes. Do we need both?

File: tests/sentry/api/endpoints/test_group_user_reports.py [hunk 2]
```diff
@@ -1,15 +1,67 @@
...
+
+    def create_events_for_environment(self, group, environment, num_events):
+        return [self.create_event(group=group, tags={
```
Comment (fpacifici) [inline tests/sentry/api/endpoints/test_group_user_reports.py]: Are you also planning to replace create_event with store_event ?

File: tests/sentry/api/endpoints/test_group_user_reports.py (hunk 1 above)
Comment (lynnagara) [reply tests/sentry/api/endpoints/test_group_user_reports.py]: Oh good catch, can probably delete this one

File: tests/sentry/api/endpoints/test_group_user_reports.py (hunk 2 above)
Comment (lynnagara) [reply tests/sentry/api/endpoints/test_group_user_reports.py]: It should be replaced, but I don't want to tackle rewriting these tests right now :)

**Review by fpacifici — APPROVED** [review]

**Pull request merged**